src/services/counters/CounterService.ts

```diff
@@ -0,0 +1,132 @@
+import {App} from "../../ManagementConsole";
+import {DmrService} from "../dmr/DmrService";
+import {ICounter} from "./ICounter";
+import {IDmrRequest} from "../dmr/IDmrRequest";
+import {SocketBindingService} from "../socket-binding/SocketBindingService";
+import IQService = angular.IQService;
+import {LaunchTypeService} from "../launchtype/LaunchTypeService";
+import {isNotNullOrUndefined, traverse, deepValue} from "../../common/utils/Utils";
+import {ICacheContainer} from "../container/ICacheContainer";
+import {CompositeOpBuilder} from "../dmr/CompositeOpBuilder";
+import {IDmrCompositeReq} from "../dmr/IDmrCompositeReq";
+import {JGroupsService} from "../jgroups/JGroupsService";
+import {IServerAddress} from "../server/IServerAddress";
+import {WeakCounter} from "./WeakCounter";
+import {StrongCounter} from "./StrongCounter";
+
+const module: ng.IModule = App.module("managementConsole.services.counter", []);
+
+export class CounterService {
+  static $inject: string[] = ["$q", "jGroupsService", "dmrService", "socketBindingService", "launchType"];
+
+  constructor(private $q: IQService, private jgroupsService: JGroupsService, private dmrService: DmrService,
+              private socketBindingService: SocketBindingService, private launchType: LaunchTypeService) {
+  }
+
+  createStrongCounter(name: string, storage: string, initialValue: number, currentValue: number, lowerBound:number, upperBound: number, dmr: any): ICounter {
+    return new StrongCounter(name, storage, initialValue, currentValue, lowerBound, upperBound, dmr);
+  }
+
+  createWeakCounter(name: string, storage: string, initialValue: number, currentValue: number, concurrency: number, dmr: any): ICounter {
+    return new WeakCounter(name, storage, initialValue, currentValue, concurrency, dmr);
+  }
+
+  getAllCounters(cacheContainer: ICacheContainer): ng.IPromise<ICounter[]> {
```

Comment:
We can create a method `doCountersExist` that returns true if a counter resource exists. This would be a simple read-resource operation on the correct DMR path, if it isn't successful, then no counters exist. You can then wrap `getAllCounters` with ^ methid. So if counters exist, `getAllCounters`, else return an empty array. 

You can then use `doCountersExist` to enable/disable the counter pages and/or tabs

Reply:
@ryanemerson have a look now please